xclim/sdba/nbutils.py

```diff
@@ -62,77 +71,65 @@ def vecquantiles(
 
 
 @njit
-def _quantile(arr, q):
-    if arr.ndim == 1:
-        out = np.empty((q.size,), dtype=arr.dtype)
-        out[:] = np.nanquantile(arr, q)
-    else:
-        out = np.empty((arr.shape[0], q.size), dtype=arr.dtype)
-        for index in range(out.shape[0]):
-            out[index] = np.nanquantile(arr[index], q)
+def _wrapper_quantile1d(arr, q):
+    out = np.empty((arr.shape[0], q.size), dtype=arr.dtype)
+    for index in range(out.shape[0]):
+        out[index] = _nan_quantile(arr[index], q)
     return out
 
 
-def quantile(da: DataArray, q: np.ndarray, dim: str | Sequence[Hashable]) -> DataArray:
-    """Compute the quantiles from a fixed list `q`.
-
-    Parameters
-    ----------
-    da : xarray.DataArray
-        The data to compute the quantiles on.
-    q : array-like
-        The quantiles to compute.
-    dim : str or sequence of str
-        The dimension along which to compute the quantiles.
-
-    Returns
-    -------
-    xarray.DataArray
-        The quantiles computed along the `dim` dimension.
-    """
-    # We have two cases :
-    # - When all dims are processed : we stack them and use _quantile1d
-    # - When the quantiles are vectorized over some dims, these are also stacked and then _quantile2D is used.
-    # All this stacking is so that we can cover all ND+1D cases with one numba function.
-
-    # Stack the dims and send to the last position
-    # This is in case there are more than one
-    dims = [dim] if isinstance(dim, str) else dim
-    tem = utils.get_temp_dimname(da.dims, "temporal")
-    da = da.stack({tem: dims})
-
-    # So we cut in half the definitions to declare in numba
-    # We still use q as the coords, so it corresponds to what was done upstream
-    if not hasattr(q, "dtype") or q.dtype != da.dtype:
-        qc = np.array(q, dtype=da.dtype)
+def _quantile(arr, q, nreduce):
+    if arr.ndim == nreduce:
+        out = _nan_quantile(arr.flatten(), q)
```

Comment:
Thanks! I must admit I didn't put much thought in the method that is used. Since the function is hidden, I think it's reasonable to expect users to know what they're doing. But would it be better to set `alpha==beta==1/3`  as a default? Do I understand correctly that this specific method `alpha==beta==1/3` is better adapted for a general context where we have no a priori info on the distribution of data? Or are you saying: "if we don't much about the data, there is no reason to deviate from the numpy standard"?

Reply:
> Do I understand correctly that this specific method alpha==beta==1/3 is better adapted for a general context where we have no a priori info on the distribution of data

Yes that's what I mean :)

Reply:
Temperature data is often normally distributed, in which case 'normal unbiased' is likely better, with alpha = beta = 3/ 8.0, but otherwise I agree.